el-patch.el

```diff
@@ -335,12 +335,13 @@ be defined permanently."
 (defun el-patch--classify-definition-type (type)
   "Classifies a definition TYPE as a `function' or `variable'.
 TYPE is a symbol `defun', `defmacro', etc."
-  (pcase type
-    ((or 'defun 'defmacro 'defsubst 'define-minor-mode)
-     'function)
-    ((or 'defvar 'defconst 'defcustom)
-     'variable)
-    (_ (error "Unexpected definition type %S" type))))
+  (let ((classify (assoc type el-patch-deftype-alist)))
+    (if classify
```

Comment:
How about `if-let` instead?

Reply:
What's the deal with `if-let` vs `if-let*`?

Reply:
They are (nearly) identical. The semantics of the star are completely unrelated to its behavior in `let` vs. `let*`.

I prefer to use `if-let` since it is shorter.